File: 2022-2023/sourcecode/lab/lab12/Dagger2BasicsJava/app/src/main/java/pl/edu/uwr/pum/dagger2basicsjava/Computer/Computer.java
package pl.edu.uwr.pum.dagger2basicsjava.Computer;

import javax.inject.Inject;

import pl.edu.uwr.pum.dagger2basicsjava.Computer.Components.CPU;
import pl.edu.uwr.pum.dagger2basicsjava.Computer.Components.Case;
import pl.edu.uwr.pum.dagger2basicsjava.Computer.Components.GPU;
import pl.edu.uwr.pum.dagger2basicsjava.Computer.Components.Motherboard;
import pl.edu.uwr.pum.dagger2basicsjava.Computer.Components.PowerSupply;

public class Computer {
    private Case computerCase;
    private GPU gpu;
    private CPU cpu;
    private Motherboard motherboard;
    private PowerSupply powerSupply;

    @Inject
    public Computer(
            Case computerCase,
            CPU cpu,
            GPU gpu,
            Motherboard motherboard,
            PowerSupply powerSupply){
        this.computerCase = computerCase;
        this.cpu = cpu;
        this.gpu = gpu;
        this.motherboard = motherboard;
        this.powerSupply = powerSupply;
    }

    public String work(){
        return "working";
    }
}

File: 2022-2023/sourcecode/lab/lab12/Dagger2BasicsKotlin/app/src/main/java/pl/edu/uwr/pum/dagger2basicskotlin/computer/Computer.kt
package pl.edu.uwr.pum.dagger2basicskotlin.computer

import pl.edu.uwr.pum.dagger2basicskotlin.computer.components.*
import javax.inject.Inject

class Computer @Inject constructor (
    private val case: Case,
    private val gpu: GPU,
    private val cpu: CPU,
    private val motherboard: Motherboard,
    private val powerSupply: PowerSupply
        ) {
    fun work(): String{
        return "working"
    }
}